worlds/zillion/logic.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict, FrozenSet, Tuple, cast, List, Counter as _Counter
+from BaseClasses import CollectionState
+from zilliandomizer.logic_components.locations import Location
+from zilliandomizer.randomizer import Randomizer
+from zilliandomizer.logic_components.items import Item, items
+from .region import ZillionLocation
+from .item import ZillionItem
+from .id_maps import item_name_to_id
+
+zz_empty = items[4]
+
+# TODO: unit tests for these
+
+
+def set_randomizer_locs(cs: CollectionState, p: int, zz_r: Randomizer) -> int:
+    """
+    sync up zilliandomizer locations with archipelago locations
+
+    returns a hash of the player and of the set locations
+    """
+    # print("in logic set locs")
+    _hash = p
+    for loc in cs.world.get_locations():
```

Comment:
If i read correctly, this is called when checking access for each location? Iterating over all world locations for each check is kinda bad. You may want to pre-generate that hash and store it to the player's world, or pass the hash directly into lambda if that seems better to you.

Reply:
Yeah, I'll look at saving each player their own list of locations.